traits/trait_types.py

```diff
@@ -3090,12 +3076,12 @@ class Instance(BaseInstance):
     def init_fast_validate(self):
         """ Sets up the C-level fast validator. """
 
-        from .has_traits import isinterface
-
-        if (self.adapt == 0) and (not isinterface(self.klass)):
+        if self.adapt == 0:
```

Comment:
This too represents a behaviour change:  if we're using a plain `Instance(ISomeInterface)` (with no adaptation requested), the previous code would call the adaptation machinery _anyway_ to check that the value adapted to itself, and then if it didn't, would go on to do the `isinstance` check and then return the value. Now we just do the instance checks and don't invoke `adapt` in this branch.